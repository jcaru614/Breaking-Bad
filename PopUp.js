import React, { useState, useEffect } from 'react';
import {View, StyleSheet, Text, TextInput, Image, ImageBackground, ScrollView, TouchableHighlight, FlatList, TouchableOpacity, SafeAreaView } from 'react-native';

export default function PopUp(props) {
    return (
        <View style={styles.popup}>
            <Text style={styles.poptitle}>{props.name}</Text>
            <Text style={styles.text} >Birthday: {props.birthday}</Text>
            <Text style={styles.text} >Status: {props.status}</Text>
            <Text style={styles.text} >Nickname: {props.nickname}</Text>
            <Text style={styles.text} >Nickname: {props.portrayed}</Text>
            <Text style={styles.text} >Occupation: {props.occupation[0]} {",", props.occupation[1]}</Text>
          {props.button}
        </View>
    )
}

const styles = StyleSheet.create({
  popup: {
    flex: 1,
    backgroundColor: '#282C35',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 5,
  },
  popPoster: {
    width: 300,
    height: 400,
    resizeMode: 'cover',
  },
  poptitle: {
    fontSize: 25,
    fontWeight: '700',
    color: 'white',
    paddingHorizontal: 5,
    padding: 10,
  },
  rating: {
    fontSize: 20,
    fontWeight: '500',
    color: 'white',
    padding: 5,
  },
  text: {
    fontSize: 20,
    fontWeight: '300',
    color: 'white',
    paddingHorizontal: 10,
    padding: 5,
  },
})
