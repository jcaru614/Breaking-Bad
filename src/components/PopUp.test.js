import 'react-native';
import PopUp from './PopUp';
import React from 'react';
import renderer from 'react-test-renderer';

describe('it works', () => {
  it('match snap', () => {
    const tree = renderer.create(<PopUp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('on call set modalVis toggles state', () => {
      it('set state to true ', () => {
          let setModalVis = false;
          const click = jest.fn();
        //   const handleClick = jest.spyOn({setModalVis:true}, "setModalVis")
          click();
        //   const handleClick = jest.spyOn(React, "useState");
          expect(click).toHaveBeenCalled();
          expect(setModalVis).toBe(false)
      });
  });
});

