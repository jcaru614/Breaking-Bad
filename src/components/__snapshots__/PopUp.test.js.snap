// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works match snap 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#282C35",
      "flex": 1,
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "700",
        "padding": 10,
        "paddingHorizontal": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "300",
        "padding": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    Birthday: 
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "300",
        "padding": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    Status: 
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "300",
        "padding": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    Nickname: 
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "300",
        "padding": 5,
        "paddingHorizontal": 10,
      }
    }
  >
    Nickname: 
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</RCTSafeAreaView>
`;
